tests/garage/tf/policies/test_gaussian_mlp_policy_with_model.py

```diff
@@ -0,0 +1,181 @@
+from garage.tf.policies import GaussianMLPPolicyWithModel
+from garage.tf.policies import GaussianMLPPolicyWithModel2
+from tests.fixtures import TfGraphTestCase
+from tests.fixtures.envs.dummy import DummyBoxEnv
+
+
+class TestGaussianMLPPolicyWithModel(TfGraphTestCase):
+    def test_gaussian_mlp_policy_with_model(self):
+        box_env = TfEnv(DummyBoxEnv())
+        policy1 = GaussianMLPPolicyWithModel(
+            env_spec=box_env, init_std=1.1, name='P1')
+        policy2 = GaussianMLPPolicyWithModel(
+            env_spec=box_env, init_std=1.2, name='P2')
+        policy3 = GaussianMLPPolicyWithModel2(
+            env_spec=box_env, init_std=1.1, name='P3')
+        policy4 = GaussianMLPPolicyWithModel2(
+            env_spec=box_env, init_std=1.2, name='P4')
+
+        # self.sess.run(tf.global_variables_initializer())
+
+        obs = [box_env.reset()]
+
+        obs_ph = tf.placeholder(tf.float32, shape=(None, 1))
+
+        ############################################################
+        # dist_info_sym
+        # API unchanged
+        #
+        # * Existing way *
+        # dist1_sym and dist2_sym are still dict(mean=mean, log_std=log_std)
+        dist1_sym = policy1.dist_info_sym(obs_ph, name='p1_sym')
+        dist2_sym = policy2.dist_info_sym(obs_ph, name='p2_sym')
+
+        # * New way *
+        # this will be tf.distributions
+        policy3.dist_info_sym(obs_ph, name='p3_sym')
+        policy4.dist_info_sym(obs_ph, name='p4_sym')
+
+        ############################################################
+        # kl_sym
+        # * Existing way *
+        kl_diff_sym1 = policy1.distribution.kl_sym(dist1_sym, dist2_sym)
+        objective1 = tf.reduce_mean(kl_diff_sym1)
+
+        kl_func = tensor_utils.compile_function([obs_ph], objective1)
+        kl1 = kl_func(obs, obs)
+
+        # * New way *
+        # equvaient to distribution.kl_sym()
+        kl_diff_sym2 = tfp.distributions.kl_divergence(
+            policy3.model.networks['default'].distribution,
+            policy4.model.networks['default'].distribution)
+        objective2 = tf.reduce_mean(kl_diff_sym2)
+
+        kl2 = self.sess.run(
+            objective2,
+            feed_dict={
+                policy3.model.networks['default'].input: obs,
+                policy4.model.networks['default'].input: obs
+            })
+
+        self.assertAlmostEqual(kl1, kl2)
+
+        ############################################################
+        # log_likelihood_sym
+        # * Existing way *
+        log_prob_sym1 = policy1.distribution.log_likelihood_sym(
+            obs_ph, dist1_sym)
+        log_prob_func = tensor_utils.compile_function([obs_ph], log_prob_sym1)
+        log_prob1 = log_prob_func(obs)
+
+        # * New way *
+        # equvaient to distribution.log_likelihood_sym(X, dist)
+        log_prob_sym2 = policy3.model.networks[
+            'default'].distribution.log_prob(
+                policy3.model.networks['default'].input)
```

Comment:
why the value is `input`? shouldn't the value be the sampled action?

Reply:
This is just illustration on how `log_likelihood_sym` will look like.